### PR TITLE
RFC: Rebase by Image Digest Reference

### PR DESCRIPTION
**Pull request by joeybrown-sf**
[Readable](https://github.com/joeybrown-sf/rfcs/blob/rebase/digest-ref/text/0000-rebase-immutable-image-ref.md)

### CODE REVIEW

File: text/0000-rebase-immutable-image-ref.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+Today, we can execute rebase by using **tag references** but not **digest references**.
+
+Here are some examples of valid rebase commands using **tag references**:
+1. `lifecycle rebase my-repo/foo`
+1. `lifecycle rebase my-repo/foo:latest`
+1. `lifecycle rebase my-repo/foo:v4`
+
+Here are some examples of currently invalid rebase commands using **digest references**:
+1. `lifecycle rebase my-repo/foo@sha256:1234 myrepo/foo`
```
Comment (natalieparellano) [inline text/0000-rebase-immutable-image-ref.md:55]: I wonder if introducing an explicit `-previous-image` flag would make things clearer:
`lifecycle rebase -previous-image my-repo/foo@sha256:1234 myrepo/foo`
If `-previous-image` is not provided we infer it from the first argument, just like `analyzer`.

Comment (joeybrown-sf) [reply text/0000-rebase-immutable-image-ref.md:55]: @natalieparellano Do you think I should reframe this RFC around this "previous-image" flag?

A new title might be `Add optional "-previous-image" flag to lifecycle rebase`.

Comment (jabrown85) [reply text/0000-rebase-immutable-image-ref.md:55]: I think an explicit argument is fine here. I agree it is an implementation detail of the RFC as well.

Comment (hone) [reply text/0000-rebase-immutable-image-ref.md:55]: :+1: this makes sense to me.

**Comment by hone** [comment]
@joeybrown-sf unless there are objections, can you update the examples with the `-previous-image` flag and also add the changes to `pack`?

**Comment by natalieparellano** [comment]
@jabrown85 would you be interested in stewarding this one?

**Comment by jabrown85** [comment]
Sure @natalieparellano! I got this one

**Comment by joeybrown-sf** [comment]
> @joeybrown-sf unless there are objections, can you update the examples with the -previous-image flag and also add the changes to pack?

@hone Are you asking for one more RFC PR? Very similar to this one, except different labels? Or something else?

**Review by natalieparellano — APPROVED** [review]

**Comment by hone** [comment]
> @hone Are you asking for one more RFC PR? Very similar to this one, except different labels? Or something else?

Just updating the current RFC.

**Review by hone — APPROVED** [review]

**Review by jabrown85 — APPROVED** [review]

**Comment by jabrown85** [comment]
Voting scheduled to end January 12th

**Pull request merged**